ERP-BACKEND/app/services/search_service.py

```
@@ -251,9 +252,18 @@ def search(
-                        )
+                        # For JSON column, use json_each or direct comparison depending on DB
+                        # Using PostgreSQL JSONB syntax with ->> operator via astext
+                        try:
+                            base_query = base_query.filter(
+                                SearchIndex.meta_data[meta_key].astext.cast(String) == str(value)
+                            )
+                        except AttributeError:
+                            # Fallback for databases that don't support astext (e.g., SQLite in tests)
+                            # Use a workaround by casting the whole JSON and comparing
+                            base_query = base_query.filter(
+                                cast(SearchIndex.meta_data[meta_key], String) == f'"{str(value)}"'
```

Ruff (0.16.1)

[warning] Use explicit conversion flag

Replace with conversion flag

(RUF010)